libgfortran/libgfortran.h

```diff
@@ -67,8 +67,14 @@ extern float __strtof (const char *, char **);
 #define gfc_strtof __strtof
 extern double __strtod (const char *, char **);
 #define gfc_strtod __strtod
+
```

Comment:
Did you try to change line 65 from:

`#ifdef __MINGW32__`

to 

`#if defined(__MINGW32__) && ! defined(__aarch64__)`

I guess, it might be a simpler fix of the issue, in case it works, ofc (haven't tried).

Reply:
Moved to a separate PR
https://github.com/ZacWalk/gcc-woarm64/pull/10